### DISCUSSION TITLE
Use universal-pathlib to support arguments with remote storage paths

### DISCUSSION THREAD

**Discussion by danielgafni** [fastapi/typer#568] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import typer
from upath import UPath

def main(write_path: UPath = typer.Argument(...)):
    with write_path.open("wb") as file:
        file.write("Hello Any Storage!")

if __name__ == "__main__":
    typer.run(main)
```

### Description

[universal-pathlib](https://github.com/fsspec/universal_pathlib) is an extremely useful `pathlib` extension that's part of the `fsspec` stack and provides the same interface as `pathlib` with support to remote filesystems such as `s3`, `gcs` and more. 

Right now `typer` doesn't understand `UPath` (and for example can't check remote file existance). 

It would be useful to support `UPath` and it would cost around 0 efforts to add this support as the interface is exactly `pathlib`'s. 

I want this to be possible out of the box:

```shell
python cli.py --write-path s3://bucket/file
```

**Comment by gshiba** [comment]
https://github.com/fastapi/typer/pull/1183

**Comment by YuriiMotov** [comment]
This is now supported, but you need to add `parser=UPath`:
```py
def main(write_path: Annotated[UPath, typer.Argument(..., parser=UPath)]):
```
